### DISCUSSION TITLE
'pysplishsplash.Exec.SimulatorBase' object has no attribute 'getScene'

### DISCUSSION THREAD

**Discussion by chunleili** [InteractiveComputerGraphics/SPlisHSPlasH#188] [Q&A] [answered]
Again, I tried the following codes in the documentation. It fails
![image](https://user-images.githubusercontent.com/48758868/174014234-143bb185-3ec2-4df3-ab8e-8e230676ac05.png)

```
import pysplishsplash
import pysplishsplash.Utilities.SceneLoaderStructs as Scene

def main():
    base = pysplishsplash.Exec.SimulatorBase()
    args = base.init()
    gui = pysplishsplash.GUI.Simulator_GUI_imgui(base)
    base.setGui(gui)
    scene = base.getScene()
    add_block = Scene.FluidBlock('Fluid', Scene.Box([0.0, 0.0, 0.0], [1.0, 1.0, 1.0]), 0, [0.0, 0.0, 0.0])
    scene.fluidBlocks[1] = add_block # In Place construction not supported yet
    base.run()

if __name__ == "__main__":
    main()
```

The error message:
```
Traceback (most recent call last):
  File "e:\codes\SPlisHSPlasH\pySPlisHSPlasH\myExamples\mySimulator.py", line 15, in <module>
    main()
  File "e:\codes\SPlisHSPlasH\pySPlisHSPlasH\myExamples\mySimulator.py", line 9, in main
    scene = base.getScene()
AttributeError: 'pysplishsplash.Exec.SimulatorBase' object has no attribute 'getScene'
```

**Comment by janbender** [comment, marked as answer]
getScene is not a function of base. The correct code should look like this:

```python
import pysplishsplash
import pysplishsplash.Utilities.SceneLoaderStructs as Scene

def main():
    base = pysplishsplash.Exec.SimulatorBase()
    args = base.init()
    gui = pysplishsplash.GUI.Simulator_GUI_imgui(base)
    base.setGui(gui)
    scene = pysplishsplash.Exec.SceneConfiguration.getCurrent().getScene()
    scene.fluidBlocks.append(Scene.FluidBlock(id='Fluid', box=Scene.Box([0.0, 0.0, 0.0], [1.0, 1.0, 1.0]), mode=0, initialVelocity=[0.0, 0.0, 0.0]))
    base.run()

if __name__ == "__main__":
    main()
```

Also take a look at the example: complex_scene.py

**Reply by chunleili** [reply]
Thank you.
I will try it.

**Reply by BoyuanTang331** [reply]
hi，it seems that there is a new problem, how could it fix

'''AttributeError: module 'pysplishsplash.Utilities.SceneLoaderStructs' has no attribute 'Box'   '''

SPlisHSPlasH version: 2.12.4  python 3.9

**Reply by BoyuanTang331** [reply]
@janbender 
<img width="963" alt="1693132157518" src="https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/117408630/a55523dd-8fcc-4414-8393-1d99dda1f9a3">

after seeing the comlex_scene.py,  Should I try Boxmax and Boxmin? but it seems to need import some files to make sure to use boxMax and boxMin to define the scene

may I ask which old version of splashsplish will run successfully, I can use an old version of the library

**Reply by janbender** [reply]
You should use boxMin and boxMax. You can find an example in the file custom_scene.py
